### PR TITLE
[PPP-4795] Added endpoint user permission check before creating DataS…

### PR DESCRIPTION
**Pull request by eddie-martinez**
…ource

### CODE REVIEW

**Comment by ddiroma** [comment]
@lucboudreau

**Comment by buildguy** [comment]
### :warning: Build finished in 44m 4s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 6 test(s) failed:

<details><summary><b>org.pentaho.platform.repository.JcrBackedDatasourceMgmtServiceTest.testCreateDatasource on testCreateDatasource(org.pentaho.platform.repository.JcrBackedDatasourceMgmtServiceTest)</b> (click to expand)</summary><p>

```
Access Denied
```

</p></details>

<details><summary><b>org.pentaho.platform.repository.JcrBackedDatasourceMgmtServiceTest.testCreateDatasource</b> (click to expand)</summary><p>

```
Access Denied
```

</p></details>

<details><summary><b>org.pentaho.platform.repository.webservices.DatasourceMgmtToWebServiceAdapterTest.testEverything on testEverything(org.pentaho.platform.repository.webservices.DatasourceMgmtToWebServiceAdapterTest)</b> (click to expand)</summary><p>

```
java.lang.RuntimeException: org.pentaho.platform.api.engine.security.userroledao.AccessDeniedException: Access Denied
```

</p></details>

<details><summary><b>org.pentaho.platform.repository.webservices.DatasourceMgmtToWebServiceAdapterTest.testEverything</b> (click to expand)</summary><p>

```
java.lang.RuntimeException: org.pentaho.platform.api.engine.security.userroledao.AccessDeniedException: Access Denied
```

</p></details>

<details><summary><b>org.pentaho.platform.repository.webservices.DefaultDatasourceMgmtWebServiceTest.testEverything on testEverything(org.pentaho.platform.repository.webservices.DefaultDatasourceMgmtWebServiceTest)</b> (click to expand)</summary><p>

```
org.pentaho.platform.api.engine.security.userroledao.AccessDeniedException: Access Denied
```

</p></details>

<details><summary><b>org.pentaho.platform.repository.webservices.DefaultDatasourceMgmtWebServiceTest.testEverything</b> (click to expand)</summary><p>

```
org.pentaho.platform.api.engine.security.userroledao.AccessDeniedException: Access Denied
```

</p></details>

  
Tests run: **2114**, Failures: **6**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-11/job/PR-5155/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by lucboudreau** [comment]
@eddie-martinez This PR seems to have broken some tests. Please review.

File: repository/src/main/java/org/pentaho/platform/repository/JcrBackedDatasourceMgmtService.java [hunk 1]
```diff
@@ -27,7 +27,9 @@
 import org.pentaho.database.model.IDatabaseConnection;
 import org.pentaho.database.service.IDatabaseDialectService;
 import org.pentaho.di.repository.RepositoryObjectType;
```
Comment (peterrinehart) [inline repository/src/main/java/org/pentaho/platform/repository/JcrBackedDatasourceMgmtService.java:29]: Looks good; update the copyright date please.

**Review by peterrinehart — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :warning: Build finished in 44m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/pentaho-data,repository
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 2 test(s) failed:

<details><summary><b>org.pentaho.platform.repository.webservices.DefaultDatasourceMgmtWebServiceTest.testEverything on testEverything(org.pentaho.platform.repository.webservices.DefaultDatasourceMgmtWebServiceTest)</b> (click to expand)</summary><p>

```
org.pentaho.platform.api.repository.datasource.DatasourceMgmtServiceException: Access Denied
```

</p></details>

<details><summary><b>org.pentaho.platform.repository.webservices.DefaultDatasourceMgmtWebServiceTest.testEverything</b> (click to expand)</summary><p>

```
org.pentaho.platform.api.repository.datasource.DatasourceMgmtServiceException: Access Denied
```

</p></details>

  
Tests run: **2116**, Failures: **2**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-11/job/PR-5155/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request closed without merging**